openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -35,6 +35,68 @@ function getJqueryElements(selector){
     return jQueryElementArray;
 }
 
+export function initRoleValidation() {
+    const dataConfig = JSON.parse(document.querySelector('#roles').dataset.config);
+    $('#roles').repeat({
+        vars: {prefix: 'edition--'},
+        validate: function (data) {
+            if (data.role == '' || data.role == '---') {
+                return error('#role-errors', '#select-role', dataConfig['Please select a role.']);
+            }
+            if (data.name == '') {
+                return error('#role-errors', '#role-name', dataConfig['You need to give this ROLE a name.'].replace(/ROLE/, data.role));
+            }
+            $('#role-errors').hide();
+            return true;
+        }
+    });
+}
+
+export function initIdentifierValidation() {
+    const dataConfig = JSON.parse(document.querySelector('#identifiers').dataset.config);
```

Comment:
consider calling this i18nConfig or translationConfig to make it clearer these map keys to translation.